FreeRTOS-Plus/Demo/AWS/Jobs_Windows_Simulator/Jobs_Demo/DemoTasks/JobsDemoExample.c

```diff
@@ -0,0 +1,838 @@
+        xJsonStatus = JSON_Search( ( char * ) pxPublishInfo->pPayload,
+                                   pxPublishInfo->payloadLength,
+                                   jobsexampleQUERY_KEY_FOR_JOB_ID,
+                                   jobsexampleQUERY_KEY_FOR_JOB_ID_LENGTH,
+                                   &pcJobId,
+                                   &ulJobIdLength );
+
+        configASSERT( ulJobIdLength < JOBS_JOBID_MAX_LENGTH );
+
+        if( xJsonStatus == JSONSuccess )
+        {
+            LogInfo( ( "Received a Job from AWS IoT Jobs service: JobId=%.*s",
+                       ulJobIdLength, pcJobId ) );
+
+            /* Process the Job document and execute the job. */
+            prvProcessJobDocument( pxPublishInfo, pcJobId, ( uint16_t ) ulJobIdLength );
+        }
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/* This is the callback function invoked by the MQTT stack when it receives
+ * incoming messages. This function demonstrates how to use the Jobs_MatchTopic
+ * function to determine whether the incoming message is a Jobs message
+ * or not. If it is, it handles the message depending on the message type.
+ */
+static void prvEventCallback( MQTTContext_t * pxMqttContext,
+                              MQTTPacketInfo_t * pxPacketInfo,
+                              MQTTDeserializedInfo_t * pxDeserializedInfo )
+{
+    uint16_t usPacketIdentifier;
+
+    ( void ) pxMqttContext;
+
+    configASSERT( pxDeserializedInfo != NULL );
+    configASSERT( pxMqttContext != NULL );
+    configASSERT( pxPacketInfo != NULL );
+
+    usPacketIdentifier = pxDeserializedInfo->packetIdentifier;
+
+    /* Handle incoming publish. The lower 4 bits of the publish packet
+     * type is used for the dup, QoS, and retain flags. Hence masking
+     * out the lower bits to check if the packet is publish. */
+    if( ( pxPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+    {
+        configASSERT( pxDeserializedInfo->pPublishInfo != NULL );
+        JobsTopic_t topicType = JobsMaxTopic;
+        char * pcJobId = NULL;
+        uint16_t usJobIdLength = 0;
+        JobsStatus_t xStatus = JobsError;
+
+        LogDebug( ( "Received an incoming publish message: TopicName=%.*s",
+                    pxDeserializedInfo->pPublishInfo->topicNameLength,
+                    ( const char * ) pxDeserializedInfo->pPublishInfo->pTopicName ) );
+
+        /* Let the Jobs library tell us whether this is a Jobs message. */
+        xStatus = Jobs_MatchTopic( ( char * ) pxDeserializedInfo->pPublishInfo->pTopicName,
+                                   pxDeserializedInfo->pPublishInfo->topicNameLength,
+                                   democonfigTHING_NAME,
+                                   THING_NAME_LENGTH,
+                                   &topicType,
+                                   &pcJobId,
+                                   &usJobIdLength );
```

Comment:
Thanks for noticing, I do use the Job ID when processing the Job document after explicitly parsing it from the JSON payload. I have updated the logic to avoid the explicit parse and instead use the parsed output from the `Jobs_MatchTopic` function

Reply:
I reverted the change (of using JobID from the `MatchTopic` function) because the `start-next` and `notify-next` topics do not contain jobID in their topics. I have updated the implementation to pass `NULL` values for these arguments